common/db/openssl/openssl.go

```diff
@@ -15,9 +16,10 @@ import (
 
 // For connecting to the database over ssl
 type SSLDBConnector struct {
-	dialInfo  *mgo.DialInfo
-	dialError error
-	ctx       *openssl.Ctx
+	dialInfo       *mgo.DialInfo
+	dialErrorMutex sync.Mutex
+	dialError      error
```

Comment:
I understand that the current logic is not correct but I still don't think it should be removed. mgo discards the error from the dialer a simply returns `"no reachable servers"` which makes it much harder to debug SSL errors. For example connecting with an invalid certificate currently errors with:
```
$ ./bin/mongodump --host localhost:27017 --db test --ssl
2017-05-23T12:46:46.735-0700	Failed: error connecting to db server: no reachable servers, openssl error: SSL errors: SSL routines:ssl3_get_server_certificate:certificate verify failed
```
But if we remove the dialError:
```
./bin/mongodump --host localhost:27017 --db test --ssl
2017-05-23T12:46:46.735-0700	Failed: error connecting to db server: no reachable servers
```